src/client/components/vision/camera/view_model.ts

```diff
@@ -0,0 +1,282 @@
+import { PlaneBufferGeometry } from 'three'
+import { RawShaderMaterial } from 'three'
+import { MeshBasicMaterial } from 'three'
+import { WebGLRenderTarget } from 'three'
+import { Scene } from 'three'
+import { Mesh } from 'three'
+import { WebGLRenderer } from 'three'
+import { Camera } from 'three'
+import { Object3D } from 'three'
+import { OrthographicCamera } from 'three'
+import { Material } from 'three'
+import { DataTexture } from 'three'
+import { LuminanceFormat } from 'three'
+import { RGBFormat } from 'three'
+import { Texture } from 'three'
+import { UnsignedByteType } from 'three'
+import { ClampToEdgeWrapping } from 'three'
+import { LinearFilter } from 'three'
+import { NearestFilter } from 'three'
+import { PlaneGeometry } from 'three'
+import { Matrix4 } from 'three'
+import { Vector4 } from 'three'
+import { Vector2 } from 'three'
+import { Geometry } from 'three'
+
+import { memoize } from '../../../base/memoize'
+
+import { CameraModel } from './model'
+import { ImageModel } from './model'
+import * as bayerFragmentShader from './shaders/bayer.frag'
+import * as bayerVertexShader from './shaders/bayer.vert'
+
+export function fourcc(code: string): number {
+  return code.charCodeAt(3) << 24 | code.charCodeAt(2) << 16 | code.charCodeAt(1) << 8 | code.charCodeAt(0)
+}
+
+export class CameraViewModel {
+  @observable.ref canvas: HTMLCanvasElement | null = null
+
+  constructor(private model: CameraModel) {
+  }
+
+  // See this createTransformer here? that means that once it is no longer observed
+  // it will be deleted. However the model that creates it will not.
+  // This means that any @computed fields will only live so long as the view does as
+  // this viewModel will only live so long as the view does.
+  static of = createTransformer((model: CameraModel) => {
+    return new CameraViewModel(model)
+  })
+
+  @computed
+  get id() {
+    return this.model.id
+  }
+
+  @computed
+  get name() {
+    return this.model.name
+  }
+
+  renderer = createTransformer((canvas: HTMLCanvasElement | null) => {
+    if (canvas) {
+      return new WebGLRenderer({ canvas })
+    }
+  }, (renderer?: WebGLRenderer) => {
+    if (renderer) {
+      renderer.dispose()
+    }
+  })
+
+  @computed
+  get camera(): Camera {
+    const camera = new OrthographicCamera(-1, 1, 1, -1, 1, 3)
+    camera.position.z = 2
+    return camera
+  }
+
+  @computed
+  get scene(): Scene {
+    const scene = new Scene()
+    if (this.model.image) {
+      scene.add(this.image(this.model.image))
+    }
+    return scene
+  }
+
+  @computed
+  get width() {
+    return this.model.image ? this.model.image.width : 1280
+  }
+
+  @computed
+  get height() {
+    return this.model.image ? this.model.image.height : 1024
+  }
+
+  private image = createTransformer((image: ImageModel): Mesh => {
+    const mesh =  new Mesh(this.quadGeometry, this.imageMaterial(image))
+    mesh.scale.y = -1 // We need to flip Y to fix the image coordinates
```

Comment:
What does `fix` mean? What is wrong with the image coordinates? Can we move any flipping to the texture loading instead?